src/raftstore/store/peer.rs

```diff
@@ -484,6 +494,43 @@ impl Peer {
         StaleState::Valid
     }
 
+    fn next_lease_expired_time(&self, send_to_quorum_ts: Timespec) -> Timespec {
+        // The valid leader lease should be
+        // "lease = election_timeout - (quorum_commit_ts - send_to_quorum_ts)"
+        // And the expired timestamp for that leader lease is "quorum_commit_ts + lease",
+        // which is "send_to_quorum_ts + election_timeout" in short.
+        send_to_quorum_ts + self.election_timeout
+    }
+
+    fn update_leader_lease(&mut self, ready: &Ready) {
+        // Update leader lease variables when the raft state changes
+        if ready.ss.is_some() {
+            let ss = ready.ss.as_ref().unwrap();
+            match ss.raft_state {
+                StateRole::Candidate => {
+                    // There is a time window between the underlying raft node receives
+                    // RequestVote responses from quorum inside raft module, and peer is notified
+                    // the new raft state is leader in in function.
+                    // This is usually caused the scheduling of OS.
+                    // The lease expired time is set here when the underlying raft node
+                    // becomes candidate, before sending out RequestVote requests.
+                    // So the leader lease is still safe for lease read if this peer becomes
+                    // leader, no matter how the scheduling of OS works.
```

Comment:
I think the last edit is safe. It's inaccurate and useless to update lease time here.